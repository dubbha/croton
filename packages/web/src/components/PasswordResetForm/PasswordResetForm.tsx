import React, { useState, FormEvent, useEffect } from 'react';
import {
  Form,
  SubmitButton,
  ErrorAlert,
  InfoAlert,
  AlertPlaceholder,
  LoadingSpinner
} from 'elements';
import './styles.scss';

type Props = {
  isLoading: boolean;
  error: string | null;
  info: string | null;
  onSubmit: (email: string) => void;
};

export const PasswordResetForm = ({
  isLoading,
  error,
  info,
  onSubmit
}: Props) => {
  const [email, setEmail] = useState('');
  const [isValid, setIsValid] = useState(false);

  useEffect(() => {
    setIsValid(!!email);
  }, [email]);

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    onSubmit(email);
  };

  return (
    <Form onSubmit={handleSubmit}>
      {error && <ErrorAlert>{error}</ErrorAlert>}
      {info && <InfoAlert>{info}</InfoAlert>}
      {!error && !info && <AlertPlaceholder height="62px" />}
      <Form.Group controlId="formEmail">
        <Form.Label>Email address</Form.Label>
        <Form.Control
          type="email"
          placeholder="Enter email"
          value={email}
          onChange={e => setEmail(e.target.value)}
          data-testid="passwordResetForm__email"
        />
      </Form.Group>

      <SubmitButton disabled={!isValid || isLoading}>
        <div className="spinner-container">
          {isLoading && <LoadingSpinner />}
        </div>
        <span>Send Email</span>
      </SubmitButton>
    </Form>
  );
};
