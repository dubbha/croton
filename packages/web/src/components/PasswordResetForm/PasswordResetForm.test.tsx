import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { PasswordResetForm } from './PasswordResetForm';

describe('components/PasswordResetForm', () => {
  const fn = jest.fn();

  const props = {
    isLoading: false,
    error: null,
    info: null,
    onSubmit: fn,
  };

  it('should render successfully', () => {
    const { container } = render(<PasswordResetForm {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render submit button disabled by default', () => {
    const { container, getByTestId } = render(<PasswordResetForm {...props} />);
    expect(container.firstChild).toMatchSnapshot();

    const submitButton = getByTestId('submitButton');
    expect(submitButton).toBeDisabled();
  });

  it('should call onSubmit prop on submit', () => {
    const { container, getByTestId } = render(<PasswordResetForm {...props} />);
    expect(container.firstChild).toMatchSnapshot();

    const emailInput = getByTestId('passwordResetForm__email');
    const submitButton = getByTestId('submitButton');

    fireEvent.change(emailInput, { target: { value: 'admin@admin.com' } });

    expect(submitButton).toBeEnabled();

    fireEvent.click(submitButton);
    expect(fn).toBeCalled();
  });
});
