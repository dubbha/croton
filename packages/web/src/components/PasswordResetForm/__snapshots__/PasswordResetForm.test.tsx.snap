// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PasswordResetForm should call onSubmit prop on submit 1`] = `
<form
  class=""
>
  <div
    data-testid="alertPlaceholder"
    style="height: 62px;"
  />
  <div
    class="form-group"
  >
    <label
      class="form-label"
      for="formEmail"
    >
      Email address
    </label>
    <input
      class="form-control"
      data-testid="passwordResetForm__email"
      id="formEmail"
      placeholder="Enter email"
      type="email"
      value=""
    />
  </div>
  <button
    class="btn btn-primary"
    data-testid="submitButton"
    disabled=""
    type="submit"
  >
    <div
      class="spinner-container"
    />
    <span>
      Send Email
    </span>
  </button>
</form>
`;

exports[`components/PasswordResetForm should render submit button disabled by default 1`] = `
<form
  class=""
>
  <div
    data-testid="alertPlaceholder"
    style="height: 62px;"
  />
  <div
    class="form-group"
  >
    <label
      class="form-label"
      for="formEmail"
    >
      Email address
    </label>
    <input
      class="form-control"
      data-testid="passwordResetForm__email"
      id="formEmail"
      placeholder="Enter email"
      type="email"
      value=""
    />
  </div>
  <button
    class="btn btn-primary"
    data-testid="submitButton"
    disabled=""
    type="submit"
  >
    <div
      class="spinner-container"
    />
    <span>
      Send Email
    </span>
  </button>
</form>
`;

exports[`components/PasswordResetForm should render successfully 1`] = `
<form
  class=""
>
  <div
    data-testid="alertPlaceholder"
    style="height: 62px;"
  />
  <div
    class="form-group"
  >
    <label
      class="form-label"
      for="formEmail"
    >
      Email address
    </label>
    <input
      class="form-control"
      data-testid="passwordResetForm__email"
      id="formEmail"
      placeholder="Enter email"
      type="email"
      value=""
    />
  </div>
  <button
    class="btn btn-primary"
    data-testid="submitButton"
    disabled=""
    type="submit"
  >
    <div
      class="spinner-container"
    />
    <span>
      Send Email
    </span>
  </button>
</form>
`;
